src/coreclr/vm/threadsuspend.cpp

```diff
@@ -1963,38 +1963,93 @@ void ThreadSuspend::UnlockThreadStore(BOOL bThreadDestroyed, ThreadSuspend::SUSP
 }
 
 
+#ifdef TARGET_X86
+#define CONTEXT_COMPLETE (CONTEXT_FULL | CONTEXT_FLOATING_POINT |       \
+                          CONTEXT_DEBUG_REGISTERS | CONTEXT_EXTENDED_REGISTERS | CONTEXT_EXCEPTION_REQUEST)
+#else
+#define CONTEXT_COMPLETE (CONTEXT_FULL | CONTEXT_DEBUG_REGISTERS | CONTEXT_EXCEPTION_REQUEST)
+#endif
+
+CONTEXT* AllocateOSContextHelper(LPVOID* contextBuffer)
+{
+    CONTEXT* pOSContext = NULL;
+
+#if defined(TARGET_X86) || defined(TARGET_AMD64)  
+    DWORD context = CONTEXT_COMPLETE;
+    BOOL supportsAVX = FALSE;
+
+    // Determine if the processor supports AVX so we could 
+    // retrieve extended registers
+    DWORD64 FeatureMask = GetEnabledXStateFeatures();
+    if ((FeatureMask & XSTATE_MASK_AVX) != 0)
+    {
+        context = context | CONTEXT_XSTATE;
+        supportsAVX = TRUE;
+    }
+
+    // Retrieve contextSize by passing NULL for Buffer
+    DWORD contextSize = 0;
+    // The initialize call should fail but return contextSize
+    BOOL success = InitializeContext(NULL, context, NULL, &contextSize);
+    _ASSERTE(!success);
+
+    // So now allocate a buffer of that size and call IntitializeContext again
+    LPVOID buffer = malloc(contextSize);
```

Comment:
Nit: In VM, the typically pattern is to `new (nothrow) BYTE[....]` instead of calling malloc directly

Reply:
I had in fact tried this, but I observed an assert saying `new []` isnt allowed here. Wonder what was different then.